src/topology/algebra/monoid/pointwise.lean

```diff
@@ -0,0 +1,32 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import topology.algebra.monoid.basic
+import order.filter.pointwise
+
+/-!
+# Pointwise filter operations on a topological monoid
+
+In this file we prove several lemmas about multiplication of filters on a topological monoid.
+-/
+
+open_locale filter topological_space pointwise
+open filter
+
+variables {M : Type*} [topological_space M]
+
+@[to_additive] lemma le_nhds_mul [has_mul M] [has_continuous_mul M] (a b : M) :
+  𝓝 a * 𝓝 b ≤ 𝓝 (a * b) :=
+by { rw [← map₂_mul, ← map_uncurry_prod, ← nhds_prod_eq], exact continuous_mul.tendsto _ }
```

Comment:
I assume you can use this to golf `nhds_mul`?

https://github.com/leanprover-community/mathlib/blob/247a102b14f3cebfee126293341af5f6bed00237/src/topology/algebra/group/basic.lean#L1289

Reply:
This seems to work:
```
begin
  refine le_antisymm _ (le_nhds_mul _ _),
  intros U,
  rw [← nhds_translation_mul_inv x, ← nhds_translation_mul_inv y, ← nhds_translation_mul_inv (x*y)],
  { rintros ⟨a, c, ⟨b, hb, ba⟩, ⟨d, hd, dc⟩, ac⟩,
    refine ⟨b ∩ d, inter_mem hb hd, assume v, _⟩,
    simp only [preimage_subset_iff, mul_inv_rev, mem_preimage] at *,
    rintros ⟨vb, vd⟩,
    refine ac ⟨v * y⁻¹, y, _, _, _⟩,
    { rw ← mul_assoc _ _ _ at vb, exact ba _ vb },
    { apply dc y, rw mul_right_inv, exact mem_of_mem_nhds hd },
    { simp only [inv_mul_cancel_right] } }
end
```